Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/connected_components.h

```diff
@@ -800,6 +802,177 @@ void keep_connected_components(PolygonMesh& pmesh
+ * \tparam OutputIterator a model of `OutputIterator`
+ *  holding objects of type `PolygonMesh`.
+ *
+ *  \tparam NamedParameters a sequence of Named Parameters
+ *
+ * \param pm the polygon mesh
+ * \param out output iterator to be filled with extracted polygon meshes.
+ * \param np an optional sequence of Named Parameters among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamBegin{edge_is_constrained_map} a property map containing the constrained-or-not status of each edge of `pm` \cgalParamEnd
+ *   \cgalParamBegin{face_index_map}
+ *     a property map containing an index for each face initialized from 0 to `num_vertices(pm)`
+ *   \cgalParamEnd
+ *   \cgalParamBegin{vertex_index_map}
+ *     a property map containing an index for each vertex initialized 0 to `num_vertices(pm)`
+ *   \cgalParamEnd
+ *   \cgalNPBegin{halfedge_index_map}
+ *     a property map containing an index for each halfedge initialized 0 to `num_halfedges(pm)`
+ *   \cgalNPEnd
+ * \cgalNamedParamsEnd
+ *
+ * \returns `out`.
+ */
+template <class PolygonMesh, class OutputIterator, class NamedParameters>
+OutputIterator split_connected_components(PolygonMesh& pm,
+                                          OutputIterator out,
+                                          const NamedParameters& np)
+{
+  typedef typename boost::mpl::if_c<CGAL::graph_has_property<PolygonMesh, CGAL::face_index_t>::value
+        , CGAL::face_index_t
+        ,CGAL::dynamic_face_property_t<std::size_t >
+        >::type FIM_def_tag; //or no _c ?
+
+  typedef typename boost::mpl::if_c<CGAL::graph_has_property<PolygonMesh, CGAL::halfedge_index_t>::value
+        , CGAL::halfedge_index_t
+        ,CGAL::dynamic_halfedge_property_t<std::size_t >
+        >::type HIM_def_tag; //or no _c ?
+
+  typedef typename boost::mpl::if_c<CGAL::graph_has_property<PolygonMesh, boost::vertex_index_t>::value
+        , boost::vertex_index_t
+        ,CGAL::dynamic_vertex_property_t<std::size_t >
+        >::type VIM_def_tag; //or no _c ?
+
+  typedef typename boost::lookup_named_param_def <
+    internal_np::edge_is_constrained_t,
+    NamedParameters,
+    internal::No_mark<PolygonMesh>//default
+  > ::type Ecm;
+
+  Ecm ecm = boost::choose_param( boost::get_param(np, internal_np::edge_is_constrained),
+                                   internal::No_mark<PolygonMesh>() );
+
+  return internal::split_connected_components_impl(
+        internal::get_map(
+          get_param(np, internal_np::face_index),
+          CGAL::dynamic_face_property_t<std::size_t >(),
+          FIM_def_tag(), pm),
+       internal::get_map(
+          get_param(np, internal_np::halfedge_index),
+          CGAL::dynamic_halfedge_property_t<std::size_t >(),
+          HIM_def_tag(), pm),
+        internal::get_map(
+          get_param(np, internal_np::vertex_index),
+          CGAL::dynamic_vertex_property_t<std::size_t >(),
+          VIM_def_tag(), pm),
+        ecm,
+        out, pm);
+
```

Comment:
The calls to get_map() are wrong: the tags are inverted, but once put in the right order, it won't compile.

Reply:
I don't get what you mean. What is the patch that produces the error?

Reply:
Nevermind, It was a problem later in the test. I think I'm almost finished :)

Reply:
Actually the problem was in the filling of id maps. In the case of a surface_mesh, the maps are const, so the put() did not compile, even though the code wasn't supposed to be called. I fixed that by moving the init in a specialization of get_map() for two dynamic_tags.